src/intersections.jl

```diff
@@ -75,6 +75,11 @@ Type `IntersectionType` in a Julia session to see the full list.
   CornerTouchingRaySegment
   OverlappingRaySegment
 
+  # line-ray intersection
+  CrossingLineRay
+  TouchingLineRay
+  OverlappingLineRay
```

Comment:
Similarly, let's follow the naming pattern and start with Ray instead of Line, so that you have CrossingRayLine.